internal/provider/aws/provider.go

```diff
@@ -471,23 +481,33 @@ func (p *awsProvider) ReconcileFirewallRules(
 			WithSecurityGroupID(securityGroupID),
 		)
 		if err != nil {
-			return status, fmt.Errorf("failed to get security group: %w", err)
+			return status, conditions, fmt.Errorf("failed to get security group: %w", err)
 		}
 	}
+
+	conditions = append(
+		conditions,
+		kmetav1.Condition{
+			Type:    v1alpha1.FirewallRuleConditionTypeSecurityGroupCreated,
+			Status:  kmetav1.ConditionTrue,
+			Reason:  v1alpha1.FirewallRuleConditionReasonSecurityGroupCreated,
+			Message: fmt.Sprintf("Security group created with nodename %s in %s", nodeName, aws.StringValue(instance.VpcId)),
+		},
+	)
+
 	securityGroupID := aws.StringValue(securityGroup.GroupId)
-	status.State = v1alpha1.FirewallRuleStateReserved
 	status.FirewallRuleID = securityGroup.GroupId
 
 	// Get the first network interface with a public IP address
 	networkInterface, err := eniWithPublicIP(instance)
 	if err != nil {
-		return status, fmt.Errorf("failed to get network interface with public IP: %w", err)
+		return status, conditions, fmt.Errorf("failed to get network interface with public IP: %w", err)
 	}
 
 	// Get all network interfaces associated with the security group
 	networkInterfaces, err := p.getNetworkInterfaces(ctx, securityGroupID)
 	if err != nil {
-		return status, fmt.Errorf("failed to list network interfaces: %w", err)
+		return status, conditions, fmt.Errorf("failed to list network interfaces: %w", err)
```

Comment:
Don't you want to set the condition v1alpha1.FirewallRuleConditionTypeNetworkInterfaceAssociated to Unknown here because of the error here ?